### PR TITLE
fix(mcp): G0.14-T6 decouple audit-replay session-id capture from MCP_REQUIRE_SESSION_ID

### LINKED ISSUE #1147: Task: G0.14-T6 decouple audit-replay session-id capture from MCP_REQUIRE_SESSION_ID enforcement — G8.2 lights up automatically (signal 11)
Parent goal: #221
Parent initiative: #1139

## Problem

`claude-rdc-hetzner-dc#697` signal 11
(`agent-session-id-never-captured-blocks-audit-replay`)
verified that the entire G8.2 audit-replay-as-tree feature
stack is **inert in the consumer's v0.6.0 deploy:**

- Migration 0014 (the column + index) is present.
- `POST /api/v1/audit/query` with `has_agent_session: true`
  returns 0 rows across the entire deploy lifetime.
- MCP `initialize` does NOT include an `Mcp-Session-Id` header
  in the response.
- `MCP_REQUIRE_SESSION_ID` env var is NOT set on the running
  pod.
- `GET /api/v1/audit/sessions/{id}/replay` returns 404 for
  every session-shaped id tried.

The whole G8.2 stack ships and works — but the upstream
session-id capture only fires when an env var is set,
which isn't called out in the v0.6.0 release notes or
`docs/RELEASING.md`. Operators discover the dependency
empirically by hitting the surface and finding no data.

Looking at the source at
[mcp/server.py:521-588](backend/src/meho_backplane/mcp/server.py#L521):

```python
def _bind_mcp_session_id(...):
    """Capture the Mcp-Session-Id header into a structlog contextvar.
    ...
    is True (MCP_REQUIRE_SESSION_ID env), a missing/empty header
    """
    if not session_id:
        _log.warning("mcp_session_id_required_but_missing")
        ...
        return 400, "invalid request: Mcp-Session-Id header is required"
    structlog.contextvars.bind_contextvars(mcp_session_id=str(session_id))
```

The gating logic conflates **capture** and **enforcement**.
The two should be independent:

- **Capture** = if the client sends `Mcp-Session-Id`, bind it
  to the contextvar so downstream audit rows carry it. **Always
  on.**
- **Enforce** = require the header; reject the request if
  missing. **Gated on `MCP_REQUIRE_SESSION_ID`.**

Decoupling these means audit-replay lights up automatically
for any client that sends the header (Claude Code does by
default per the consumer's `RDC Wall W7` reference in their
config), without operators needing to flip
`MCP_REQUIRE_SESSION_ID` and risk breaking older clients.

## Current state

- `_bind_mcp_session_id` at [mcp/server.py:521](backend/src/meho_backplane/mcp/server.py#L521):
  binds only when `MCP_REQUIRE_SESSION_ID` is set; otherwise
  the contextvar stays unbound and `audit_log.agent_session_id`
  stays NULL.
- Audit replay feature stack (column, route, recursive-CTE,
  MCP tool, CLI verb) ships and works but has nothing to query
  on a default deploy.

## Desired state

**Part A — decouple capture from enforcement.**
At [mcp/server.py:521](backend/src/meho_backplane/mcp/server.py#L521):

```python
def _bind_mcp_session_id(request, settings):
    session_id = request.headers.get("Mcp-Session-Id")
    if session_id:
        # ALWAYS bind when present — independent of MCP_REQUIRE_SESSION_ID
        structlog.contextvars.bind_contextvars(mcp_session_id=str(session_id))
        return  # captured

    # Missing header path — only enforce when env says so
    if settings.mcp_require_session_id:
        _log.warning("mcp_session_id_required_but_missing")
        return _http_400("invalid request: Mcp-Session-Id header is required")
    # else: capture-only-mode; no header = no agent_session_id, OK
```

**Part B — surface capture state.**
Add a `mcp_session_id_capture: "always" | "enforced"` field to
`/api/v1/health` (or `/ready`, or `/api/v1/auth-config`) so
operators see at a glance whether their deploy is capturing
session ids. This is the sister surface to T7 of this
Initiative (the `/ready` features block); fold into T7's
implementation if it's shipping the same surface, otherwise
own it here.

**Part C — release notes / runbook.**
Add a single-sentence note to `docs/RELEASING.md`'s post-deploy
section: "audit replay (G8.2) auto-lights-up when MCP clients
send `Mcp-Session-Id` — no env var required after this fix.
Older clients that don't send the header continue to work; audit
rows just won't carry `agent_session_id`."

## Acceptance criteria

- [ ] `_bind_mcp_session_id` captures whenever the header is
  present, regardless of `MCP_REQUIRE_SESSION_ID`.
- [ ] When the header is missing AND
  `MCP_REQUIRE_SESSION_ID=true`, the existing 400 enforcement
  is unchanged.
- [ ] When the header is missing AND
  `MCP_REQUIRE_SESSION_ID=false`/unset, the request proceeds
  without an `agent_session_id` (no 400, no warning).
- [ ] Tests cover all four combinations
  (header present/absent × enforce on/off).
- [ ] `/api/v1/health` or sibling surface exposes the
  capture-state.
- [ ] `docs/RELEASING.md` documents the post-deploy
  audit-replay enablement story.
- [ ] After the fix, a fresh MCP session from any client that
  sends `Mcp-Session-Id` produces audit rows with non-null
  `agent_session_id`; `GET /api/v1/audit/sessions/{id}/replay`
  returns the recursive-CTE tree.
- [ ] `Python (ruff + mypy + pytest)` green.

## Out of scope

- The G8.2 replay route itself — already shipped (#1012); this
  Task only fixes the upstream capture gating.
- Audit-row schema changes — column already exists.
- The broader feature-gate visibility story — T7 of this
  Initiative owns the `/ready` features block surface; this
  Task can fold its capture-state into T7's surface if T7 is
  ready, otherwise own a narrow `mcp_session_id_capture` field
  on `/health`.

## References

- Consumer feedback: `claude-rdc-hetzner-dc#697` signal 11
  (`agent-session-id-never-captured-blocks-audit-replay`).
- Prior work: #1010 G8.2-T2 (`_bind_mcp_session_id` originally
  landed), #1012 G8.2-T4 (`/audit/sessions/{id}/replay` route).
- Grounding:
  [mcp/server.py:521-588](backend/src/meho_backplane/mcp/server.py#L521),
  [mcp/audit.py:96-212](backend/src/meho_backplane/mcp/audit.py#L96).
- Sibling Task: **T7 of this Initiative** (/ready features
  block) — capture-state can fold here.
- Best-practices: **python_best_practices** pack (separate
  concerns: capture vs enforcement) + **devops_best_practices**
  pack (feature-gate state must be observable at deploy time).

### PR DESCRIPTION
**Pull request by damir-topic**
## Summary

- `_bind_mcp_session_id` (`backend/src/meho_backplane/mcp/server.py`) now captures `Mcp-Session-Id` unconditionally; `MCP_REQUIRE_SESSION_ID` strictly gates enforcement (missing-header reject) only. G8.2 audit replay therefore lights up on any default deploy whose MCP clients include the header — no env var flip required.
- Removed the per-call uuid4 fallback that synthesized a unique "session" id for every header-less call (which polluted the G8.2 session search with one-row sessions). Header-less / malformed-header calls now land `agent_session_id` as NULL; the replay CTE walks NULLs out naturally.
- New `HealthResponse.mcp_session_id_capture: "always" | "enforced"` (sourced from a new `mcp_session_id_capture_mode()` helper) surfaces the deploy mode on `GET /api/v1/health`. T7 #1148's `/ready` features block reads from the same helper for the `audit_replay` block (parallel-wave coordination).
- `docs/RELEASING.md` post-deploy step adds a single-sentence audit-replay auto-enablement note. `docs/codebase/audit_query.md` updates session-id capture semantics. `docs/codebase/error-message-shape.md` audit-table row added (signal 11 → compliant via the new field).

## Test plan

- [x] `ruff check` — clean on touched files
- [x] `ruff format --check` — clean
- [x] `mypy` — clean on whole package (339 source files)
- [x] `pytest tests/test_mcp_audit.py tests/test_mcp_server.py tests/test_api_v1_health.py tests/test_mcp_tool_audit_replay.py tests/test_settings.py tests/test_mcp_tool_meho_status.py tests/acceptance/test_audit_replay_acceptance.py -q` — 95 passed, 19 skipped (the skips are vacuous-skipped contract tests guarded on `MEHO_INTEGRATION_TESTS`)
- [x] `code-quality.py --diff` — 0 blockers, 9 warnings (all pre-existing)
- [x] AC: `_bind_mcp_session_id` captures whenever header is present, regardless of `MCP_REQUIRE_SESSION_ID` → `test_session_header_propagates_to_agent_session_id` + `test_mcp_session_id_capture_default_always` cover.
- [x] AC: missing header + `MCP_REQUIRE_SESSION_ID=true` → unchanged 400 → `test_require_session_id_rejects_missing_header_without_audit_row` covers.
- [x] AC: missing header + env unset → request proceeds, no 400 → `test_absent_session_header_leaves_agent_session_id_null` covers (also asserts the column is NULL, not a synthetic uuid4).
- [x] AC: all four combinations (header present/absent × enforce on/off) → covered by the four `test_*_session*` tests plus the new `test_require_session_id_with_malformed_header_succeeds_with_null`.
- [x] AC: `/api/v1/health` exposes capture state → `test_mcp_session_id_capture_default_always` + `test_mcp_session_id_capture_enforced_when_env_set`.
- [x] AC: `docs/RELEASING.md` documents the audit-replay enablement story → `### 6. Deploy + smoke` section, line added.
- [x] AC: production behavior — non-NULL `agent_session_id` for any client that sends the header; G8.2 `/api/v1/audit/sessions/{id}/replay` returns the recursive-CTE tree → verified via the existing `tests/acceptance/test_audit_replay_acceptance.py` suite (uses `structlog.contextvars.bound_contextvars(mcp_session_id=...)` to simulate the header-bound contextvar — unchanged contract).

## Out of scope

- T7 #1148's `/ready` features block + Keycloak admin error symmetry — sibling task running in parallel. This PR exposes `mcp_session_id_capture_mode()` for T7 to call; T7 reads it.
- G8.2 replay route itself (#1012) — already shipped; this Task only fixes the upstream capture gating.
- Audit-row schema changes — column already exists (migration 0014).

Closes #1147

### CODE REVIEW

**Comment by damir-topic** [comment]
@coderabbitai review

File: backend/src/meho_backplane/api/v1/health.py [hunk 1]
```diff
@@ -117,13 +118,35 @@ class DbStatus(BaseModel):
...
+    :func:`~meho_backplane.mcp.server.mcp_session_id_capture_mode`
+    helper so both surfaces stay consistent).
+    """
 
     model_config = ConfigDict(frozen=True)
 
     operator: OperatorIdentity
     vault: VaultStatus
     db: DbStatus
+    mcp_session_id_capture: str
```
Comment (damir-topic) [inline backend/src/meho_backplane/api/v1/health.py:149]: **[🛑 B1]** — CLI API snapshot is stale; the new `HealthResponse.mcp_session_id_capture` field is not reflected in `cli/api/openapi.json` or `cli/internal/api/client.gen.go`.

**What:** This PR adds `mcp_session_id_capture: str` to `HealthResponse` (line 149 here), but does not regenerate the CLI's snapshotted OpenAPI document or the generated Go client. The Go CLI's `meho status` command consumes `HealthResponse` via the typed client at `cli/internal/api/client.gen.go`.

**Why it matters:** CI's `Python (ruff + mypy + pytest) / CLI API snapshot freshness` step runs `cd cli && make snapshot-openapi && make generate` and diffs the result against HEAD. The diff (visible in the failing job's log) shows the new field would be added to both files. The check is **FAILURE** on this PR, blocking merge.

**Required change:** From the repo root, run:

```
cd cli && make snapshot-openapi && make generate
git add cli/api/openapi.json cli/internal/api/client.gen.go
git commit -s -m "fix(cli): regenerate openapi snapshot + Go client for HealthResponse.mcp_session_id_capture (G0.14-T6 #1147)"
```

then push. The CI step is enforced by the workflow at `.github/workflows/ci.yml`'s `CLI API snapshot freshness` job — running the same two `make` targets locally will reproduce the diff CI emits.

**Review by damir-topic — COMMENTED** [review]
## /auto-review-pr summary (iteration 1)




**Anchor task:** #1147 — G0.14-T6 decouple audit-replay session-id capture from `MCP_REQUIRE_SESSION_ID` enforcement
**Surface area:** backend
**Best-practices packs audited:** python_best_practices
**Verdict:** REQUEST_CHANGES
**Head SHA reviewed:** `ae8b900c57437bfe07aae76e74b8f1c1bb50d49f`
**AI reviewer (CodeRabbit):** not received — CodeRabbit posted a "Review limit reached" notice at 19:23:44Z (org credits exhausted; next review at ~20:16Z, outside this iteration's 15-minute window).

### Acceptance criteria
- [x] `_bind_mcp_session_id` captures whenever the header is present — `mcp/server.py:613-633` reads the header first, binds unconditionally on a parseable UUID. Covered by `test_session_header_propagates_to_agent_session_id` + `test_mcp_session_id_capture_default_always`.
- [x] Missing header + `MCP_REQUIRE_SESSION_ID=true` → unchanged 400 → `test_require_session_id_rejects_missing_header_without_audit_row`.
- [x] Missing header + env unset → request proceeds, no 400, column NULL → `test_absent_session_header_leaves_agent_session_id_null`.
- [x] All four combinations (header present/absent × enforce on/off) — covered, plus a new `test_require_session_id_with_malformed_header_succeeds_with_null`.
- [x] `/api/v1/health` surfaces capture state — `HealthResponse.mcp_session_id_capture` added, sourced from `mcp_session_id_capture_mode()`. Two new integration tests cover both modes.
- [x] `docs/RELEASING.md` documents the post-deploy enablement story.
- [x] After the fix, audit rows carry `agent_session_id`; replay returns the CTE tree — `tests/acceptance/test_audit_replay_acceptance.py` (uses the `mcp_session_id` contextvar binding, unchanged contract) still passes.
- [❌] `Python (ruff + mypy + pytest)` lane is green; **the `CI / CLI API snapshot freshness` lane is FAILURE** — see B1.

### Project-board alignment
| Issue | Status | Link | Auto-close | Delivered | Action |
|---|---|---|---|---|---|
| #1147 | In Progress | Closes | auto | full | close on merge |

No undisclosed linked-but-not-delivered or delivered-but-not-linked issues.

### Predicted vs. actual diff
- ✅ `backend/src/meho_backplane/mcp/server.py` — decouple + helper + `__all__` export.
- ✅ `backend/src/meho_backplane/api/v1/health.py` — new field + helper call.
- ✅ `backend/src/meho_backplane/mcp/tools/audit.py` — docstring updated to match the new unbound-on-no-header mechanism (the `_resolve_self_session_id` assertion stays correct because an unbound contextvar is `None` and the equality check fails).
- ✅ `backend/src/meho_backplane/settings.py` — docstring rewritten to spell out capture vs. enforcement.
- ✅ Tests — three existing tests rewritten for NULL semantics, three new tests added (require-mode malformed header, default-mode capture, enforced-mode capture).
- ✅ Docs — CHANGELOG, RELEASING, audit_query.md, error-message-shape.md.
- 🕳️ **Missing — `cli/api/openapi.json` + `cli/internal/api/client.gen.go` not regenerated for the new `HealthResponse.mcp_session_id_capture` field. See B1.**

### Integration verification
- [x] `ruff check src/` — clean (in worktree).
- [x] `ruff format --check src/` — 339 files already formatted.
- [x] `mypy src/` — 339 files, no issues.
- [x] `pytest tests/test_mcp_audit.py tests/test_mcp_server.py tests/test_api_v1_health.py tests/test_mcp_tool_audit_replay.py tests/test_settings.py tests/test_mcp_tool_meho_status.py tests/acceptance/test_audit_replay_acceptance.py -q` — 95 passed, 19 skipped (skips are `MEHO_INTEGRATION_TESTS`-guarded contract tests).
- [⚠️ not run locally; **CI caught it**] `cd cli && make snapshot-openapi && make generate` — `CI / CLI API snapshot freshness` is FAILURE on this PR.
- [⚠️ infra-failure, not blocking] `image / Build (and conditionally push) backplane image` — FAILURE due to docker.io anonymous pull rate limit (`429 Too Many Requests` on the python base image manifest). Not PR-related; rerun the workflow once the rate limit clears.

### Findings (inline — each is a merge blocker per team policy)
For the `/auto-implement-issue --continue-pr` punch list:

- 🛑 B1 — CLI API snapshot stale; regenerate `cli/api/openapi.json` and `cli/internal/api/client.gen.go` to reflect the new `HealthResponse.mcp_session_id_capture` field. Required change: from the repo root, run `cd cli && make snapshot-openapi && make generate`, then commit both regenerated files (CI's `CI / CLI API snapshot freshness` step runs the same two commands and diffs the result against HEAD).

### Adjacent improvements (non-blocking, not posted inline)
- `test_tools_call_meho_status_returns_health_response_shape` (`backend/tests/test_mcp_tool_meho_status.py:80`) does not pin the new `mcp_session_id_capture` field in the asserted payload. Not a regression — the chassis `HealthResponse` model serializes the new field automatically, and a dedicated test on the HTTP route (`test_mcp_session_id_capture_default_always`) covers it — but a one-line addition (`assert payload["mcp_session_id_capture"] == "always"`) would make the `meho.status` MCP-tool surface explicitly contracted on the new field too, mirroring how the test already pins `payload["db"]["migrated"] is True` for the sibling `db` field.

### AI reviewer (CodeRabbit) cross-check
- CodeRabbit returned a `Review limit reached` notice at 19:23:44Z citing exhausted organization usage credits; next review window opens ~20:16Z (53 min from rate-limit start), outside this iteration's 15-minute cap.
- Adopted: 0 — Disputed: 0 — Out of scope: 0

### Docs
- [x] `docs/codebase/audit_query.md` — updated to remove the "synthetic uuid4 fallback" prose and document the capture-vs-enforcement decouple.
- [x] `docs/codebase/error-message-shape.md` — new audit-table row for signal 11 → compliant via `mcp_session_id_capture` field.
- [x] `docs/RELEASING.md` — post-deploy section gains the audit-replay auto-enablement bullet.
- [x] `CHANGELOG.md` — new entry under [Unreleased].

### What worked well (review-body only — never posted as inline comments)
- The split between `_bind_mcp_session_id` (capture + enforcement gate) and the new `mcp_session_id_capture_mode()` helper (operator-facing introspection) is the right shape — both `/api/v1/health` (this PR) and `/ready` features block (sibling #1148) read the same helper, so the two surfaces stay consistent by construction rather than by convention.
- Dropping the per-call `uuid4()` fallback is the correct call: the prior behavior produced non-NULL but uncorrelated audit rows, which silently broke the G8.2 replay search surface in a way that *looked* superficially OK (`agent_session_id IS NOT NULL` rows existed). The new NULL-when-absent semantics is honest about what the data means.
- The require-mode + malformed-header path is well-reasoned and tested: the require contract is satisfied at the transport layer by the header's presence, the malformed value lands NULL, and a structured `mcp_malformed_session_id` warning makes the misbehaving client observable without breaking its retry loop.
- `audit_query/replay.py`'s recursive CTE walks NULLs out of the session forest by way of the `agent_session_id = :session_id` predicate (NULL `=` anything is `unknown` → false in WHERE), so no replay-engine change is needed — confirmed at `backend/src/meho_backplane/audit_query/replay.py:120-127`.
- Documentation is thorough: CHANGELOG, RELEASING, audit_query.md, error-message-shape.md all updated coherently.

**Comment by damir-topic** [comment]
@coderabbitai review

**Review by damir-topic — COMMENTED** [review]
## /auto-review-pr summary (iteration 2)




> **Note on review event state.** Authoritative verdict is **APPROVE**. GitHub forbids an own-author `APPROVE` review event (`Review Can not approve your own pull request`), so this review event is posted as `COMMENTED`. The orchestrator must read this review body and the result file at `.claude/auto/review-results/pr-1174-iter-2.json` (where `verdict: "APPROVE"`) rather than the review event state — same workaround as iter-1.

**Anchor task:** #1147 — G0.14-T6 decouple audit-replay session-id capture from `MCP_REQUIRE_SESSION_ID` enforcement
**Surface area:** backend
**Best-practices packs audited:** python
**Verdict:** APPROVE
**Head SHA reviewed:** `faf483a25adf0c57ea0033a4f60c375e2718a3cb`
**AI reviewer (CodeRabbit):** not received within 15 minutes (org credits / rate-limit; status check posted as `pass` but no review object on the new commit — re-triggered at 19:57:41Z, polled through ~22 minutes total, no findings landed; treated as unavailable for this iteration per the skill spec)

### Acceptance criteria
- [x] `_bind_mcp_session_id` captures whenever the header is present, regardless of `MCP_REQUIRE_SESSION_ID` — verified by `test_session_header_propagates_to_agent_session_id` + `test_mcp_session_id_capture_default_always`.
- [x] Missing header + `MCP_REQUIRE_SESSION_ID=true` → 400 unchanged — verified by `test_require_session_id_rejects_missing_header_without_audit_row`.
- [x] Missing header + env unset → proceed with NULL session id — verified by `test_absent_session_header_leaves_agent_session_id_null` (renamed from the prior `..._gets_fresh_uuid_per_call`, with the assertion flipped from "non-NULL distinct" to "all NULL").
- [x] All four combinations + malformed-header-in-enforce-mode → covered (new `test_require_session_id_with_malformed_header_succeeds_with_null`).
- [x] `/api/v1/health` exposes capture state via new `mcp_session_id_capture: "always" | "enforced"` field — verified by `test_mcp_session_id_capture_default_always` + `test_mcp_session_id_capture_enforced_when_env_set`.
- [x] `docs/RELEASING.md` documents the audit-replay enablement story — line landed in the post-deploy checklist (`### 6. Deploy + smoke`).
- [x] Non-NULL `agent_session_id` for header-bearing clients; recursive-CTE replay walks → contract covered by the unchanged `tests/acceptance/test_audit_replay_acceptance.py` suite.
- [x] `Python (ruff + mypy + pytest)` green — CI lane SUCCESS.

### Project-board alignment
| Issue | Status | Link | Auto-close | Delivered | Action |
|---|---|---|---|---|---|
| #1147 | In Progress | Closes | yes | yes | close on merge |

Parent linkage (body anchors): Parent goal #221, parent initiative #1139 — both unchanged from iter-1 and still consistent with PR title `G0.14-T6` and the issue body's `Parent initiative: #1139` reference. No board-side mismatch.

### Predicted vs. actual diff
- ✅ `backend/src/meho_backplane/mcp/server.py` — decouple logic + `mcp_session_id_capture_mode()` helper.
- ✅ `backend/src/meho_backplane/api/v1/health.py` — `HealthResponse.mcp_session_id_capture` field, populated via helper.
- ✅ `backend/src/meho_backplane/mcp/tools/audit.py` — docstring on `_resolve_self_session_id` updated to reflect "contextvar unbound when header absent" (no synthetic uuid4 to compare against).
- ✅ `backend/src/meho_backplane/settings.py` — docstring on `mcp_require_session_id` field clarifies capture-vs-enforcement split.
- ✅ `backend/tests/test_mcp_audit.py`, `test_mcp_server.py`, `test_api_v1_health.py`, `test_mcp_tool_audit_replay.py` — tests updated for the new contract.
- ✅ `docs/RELEASING.md`, `docs/codebase/audit_query.md`, `docs/codebase/error-message-shape.md` — doc updates.
- ✅ `CHANGELOG.md` — `[Unreleased]` entry.
- ✅ `cli/api/openapi.json` + `cli/internal/api/client.gen.go` — regenerated for the new `HealthResponse.mcp_session_id_capture` field (commit `faf483a`, the iter-1 fix).

### Integration verification
- [x] `ruff check src/meho_backplane/{api/v1/health.py,mcp/server.py,mcp/tools/audit.py,settings.py}` — clean
- [x] `ruff format --check` on the same four files — already formatted
- [x] `mypy` on the same four files — `Success: no issues found in 4 source files`
- [x] `pytest tests/test_mcp_audit.py tests/test_mcp_server.py tests/test_api_v1_health.py tests/test_mcp_tool_audit_replay.py tests/test_settings.py tests/test_mcp_tool_meho_status.py` — **95 passed**, 0 failed
- [x] `cd cli && make snapshot-openapi && make generate && git diff --exit-code cli/` — **clean** (re-ran locally; the iter-1 B1 is fully resolved)
- [x] CI `CLI API snapshot freshness` lane — **SUCCESS** (1m11s)
- [x] CI `Python (ruff + mypy + pytest)` — **SUCCESS** (4m28s)
- [x] CI `Python (integration testcontainers)` — **SUCCESS** (8m8s; the acceptance test `tests/acceptance/test_audit_replay_acceptance.py` runs in this lane with `MEHO_INTEGRATION_TESTS=1`)
- [x] CI `Go (golangci-lint + go test)` — **SUCCESS**
- [x] CI `Helm (lint + template + kubeconform)` — **SUCCESS**
- [x] CI `Build backplane image` — **SUCCESS** (12m15s; the docker.io rate-limit that flaked iter-1 is no longer firing)
- [x] CI `CodeQL`, `Semgrep OSS`, `Semgrep SAST`, `TruffleHog Secret Scan`, `DCO` — all **SUCCESS**
- [⚠️ skipped] `pr-smoke` (deploy to rke2-infra) — `skipping` by workflow policy; not an integration gap for this PR.

### Findings (inline — each is a merge blocker per team policy)
None. The original iter-1 inline blocker B1 (file=`backend/src/meho_backplane/api/v1/health.py`, line 149) remains posted as an unresolved review thread, but the code change resolving it (commit `faf483a` regenerated `cli/api/openapi.json` + `cli/internal/api/client.gen.go`) has landed; CI proves the lane is now green. The author can mark the thread resolved before merge.

### Adjacent improvements (non-blocking, not posted inline)
Items the author may pick up in a future PR. None block merge.

- `test_tools_call_meho_status_returns_health_response_shape` (`backend/tests/test_mcp_tool_meho_status.py:80`) does not pin the new `mcp_session_id_capture` field in the asserted payload. A one-line addition (`assert payload["mcp_session_id_capture"] == "always"`) would contract the MCP-tool surface on the new field, mirroring how the test already pins `db.migrated`. Not a blocker because the chassis `HealthResponse` model serializes the field automatically and a dedicated HTTP-route test (`test_mcp_session_id_capture_default_always`) covers the field's wire shape. Same adjacent observation as the iter-1 review; the iter-1 fix iteration's report explicitly punted it as out-of-scope for the B1-only fix.

### AI reviewer (CodeRabbit) cross-check
- Adopted: 0
- Disputed: 0
- Out of scope: 0

CodeRabbit's last action on this PR (at 19:23:44Z) reported `Review limit reached` (org credits exhausted). A re-trigger at 19:57:41Z was accepted (`Actions performed: Review triggered`), but no review object materialised within the configured 15-minute poll window (polled through ~22 minutes total). The `CodeRabbit` *commit-status* check shows `pass` ("Review completed") — that is a status-API stub, not a review object. Per the skill spec, `bot_review.received: false` is recorded and the verdict proceeds on Claude's own analysis. Findings on the original commit (none — CodeRabbit was rate-limited at the time) are unchanged.

### Docs
- [x] `docs/codebase/audit_query.md` — updated (session-id capture semantics now reflect decoupled capture-vs-enforcement contract; the `mcp.audit.replay` reverse-dependencies block calls out the G0.14-T6 #1147 fix path and the `mcp_session_id_capture` health field as the observability surface).
- [x] `docs/codebase/error-message-shape.md` — audit-table row added for signal 11 (`GET /api/v1/health` capture-state field — compliant via the new field).
- [x] `docs/RELEASING.md` — post-deploy step added (audit-replay auto-enablement note + `mcp_session_id_capture` field inspection).
- [n/a] No `docs/codebase/<area>.md` entry was invalidated by this PR.

### What worked well (review-body only — never posted as inline comments)
- The iter-1 fix is mechanically minimal: only `cli/api/openapi.json` and `cli/internal/api/client.gen.go` change in commit `faf483a`, exactly the two files the iter-1 B1 named, with no scope creep.
- Test coverage for the decouple is genuinely exhaustive — the four header×enforce-state combinations land as four named tests, plus the additional `test_require_session_id_with_malformed_header_succeeds_with_null` for the require-mode-malformed edge case. The renaming of `test_absent_session_header_gets_fresh_uuid_per_call` → `test_absent_session_header_leaves_agent_session_id_null` (with assertion flip) is the right shape for the new contract.
- The `mcp_session_id_capture_mode()` helper is the right factoring for T7 (sibling #1148) to consume — single source of truth for "what mode is this deploy in", surfaced on both `/api/v1/health` (this PR) and `/ready` (T7).
- Docstring updates on `_bind_mcp_session_id`, `Settings.mcp_require_session_id`, and `_resolve_self_session_id` are all internally consistent: each now references G0.14-T6 #1147 and explains the capture-vs-enforcement split clearly.
- CHANGELOG `[Unreleased]` entry is detailed and accurately reflects the operator-facing impact (auto-enablement + the new health field).
